crates/router/src/connector/iatapay/transformers.rs

```diff
@@ -212,10 +212,14 @@ impl<F, T>
         item: types::ResponseRouterData<F, IatapayPaymentsResponse, T, types::PaymentsResponseData>,
     ) -> Result<Self, Self::Error> {
         let form_fields = HashMap::new();
-        let id = match item.response.iata_payment_id {
+        let id = match item.response.iata_payment_id.clone() {
             Some(s) => types::ResponseId::ConnectorTransactionId(s),
             None => types::ResponseId::NoResponseId,
         };
+        let mut connector_response_reference_id = item.response.iata_payment_id.clone();
+        if connector_response_reference_id.is_none() {
+            connector_response_reference_id = item.response.merchant_payment_id.clone();
+        }
```

Comment:
This is no longer needed as you have already assigned the same in line 232. Please remove this
```suggestion
```